### ISSUE TITLE
Reflect.construct doesn't match spec exactly (Date constructor edge case)

### ISSUE THREAD

**Issue by Ginden** [zloirock/core-js#87]
Following code will fail with this polyfill:

```
assert(typeof Reflect.construct(Date, [2,3,4]) === 'object');
```

This problem is solvable trough native (not polyfilled) `.bind` function:
```
function construct(Constructor, args) {
var P = Constructor.bind(null, ...args);
return new P();
}
```
For enviroments without native `.bind` (IE8)  it's necessary to fallback into eval:
```
function construct(Constructor, args) {
     var f = Function('Constructor', 'return new Constructor('+args.map((e,i)=>'arguments['+(i+1)+']').join(', ')+')');
     return f(Constructor, ...args);
}
```

**Comment by zloirock** [comment]
Fallback for native constructors was added some days ago, https://github.com/zloirock/core-js/commit/af092c9eaed9a25ce5f8cf07f46a663089410494, available in master branch and will be available in 1.0. Don't think `eval`-based fallback for old IE is a good idea.

**Comment by zloirock** [comment]
Will be supported in old browsers too.
